index.js

```diff
@@ -6,19 +6,33 @@ const isType = (type, inp) => typeof inp === type
 const isFunc = inp => isType('function', inp)
 const isNum = inp => isType('number', inp)
 const isIgnored = inp => isType('undefined', inp) || isType('object', inp)
+const wrp = (pf, arg, sf) => pf.concat([...arg].concat(sf))
 
 function stringHelper () {
+  let aux = []
   let buffer = []
+  let prefixes = []
+  let suffixes = []
+  let decorators = false
+  let that = this
 
-  this.cat = function () {
+  function ccat () {
     const len = arguments.length
+    let args = arguments
     for (let i = 0; i < len; i += 1) {
-      let arg = arguments[i]
-      if (isFunc(arg)) this.cat(arg.apply(this))
-      else if (isArr(arg)) this.cat(...arg.flatten())
+      let arg = args[i]
+      if (isFunc(arg)) ccat(arg.apply(that))
```

Comment:
the `apply` shall be used when you don't know the amount of arguments and want to send an array, in this case you know there are NO arguments, use `call` instead.